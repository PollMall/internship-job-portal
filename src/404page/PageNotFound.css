.text-center {
  text-align: center;
}

.center-block {
  display: block;
  width: fit-content;
  margin: 0 auto;
}