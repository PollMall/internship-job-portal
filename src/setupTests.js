// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import { mockApolloData } from './queries/MockApolloData';

Object.defineProperty(window, 'matchMedia', {
  value: () => ({
    matches: false,
    addListener: () => {},
    removeListener: () => {},
  }),
});

Object.defineProperty(window, 'alert', {
  value: jest.fn()
})

function makeMockApolloCalls(props) {
  return props.map((p) => {
    const { query, variables, data} = p;
    console.log(data);
    return {
      request: {
        query,
        variables: variables || {}
      },
      result: {
        data: data || {}
      }
    }
  })
}

const mockApolloCalls = makeMockApolloCalls(mockApolloData);

const AllProviders = ({ children }) => {
  return (
    <MockedProvider mocks={mockApolloCalls} addTypename={false}>
       <MemoryRouter>
         {children}
       </MemoryRouter>
     </MockedProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllProviders, ...options })


export { customRender as render };