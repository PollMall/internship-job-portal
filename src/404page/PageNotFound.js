import React from 'react';
import { Link } from 'react-router-dom';
import './PageNotFound.css';

function PageNotFound() {
  return (
    <div className="display-flex">
      <h1 className="text-center">Page Not Found</h1>
      <p className="text-center">It looks like you&apos;re lost. There is nothing here.</p>
      <Link to="/" className="center-block text-center" data-testid="page-not-found-link">Go back to main page.</Link>
    </div>
  );
}

export default PageNotFound;
