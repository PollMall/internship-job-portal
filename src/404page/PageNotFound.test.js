import React from 'react';
import { render } from '../setupTests';
import PageNotFound from './PageNotFound';

describe('PageNotFound test', () => {
  it('should match snapshot', () => {
    const { container } = render(<PageNotFound />);
    expect(container).toMatchSnapshot();
  });
});
