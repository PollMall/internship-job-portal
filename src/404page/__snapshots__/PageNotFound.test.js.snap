// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound test should match snapshot 1`] = `
<div>
  <div
    class="display-flex"
  >
    <h1
      class="text-center"
    >
      Page Not Found
    </h1>
    <p
      class="text-center"
    >
      It looks like you're lost. There is nothing here.
    </p>
    <a
      class="center-block text-center"
      data-testid="page-not-found-link"
      href="/"
    >
      Go back to main page.
    </a>
  </div>
</div>
`;
